akka-http/src/main/scala/akka/http/javadsl/server/directives/FormFieldDirectives.scala

```diff
@@ -0,0 +1,72 @@
+package akka.http.javadsl.server.directives
+
+import akka.http.scaladsl.server.{ Directives ⇒ D }
+import akka.http.javadsl.server.JavaScalaTypeEquivalence._
+import akka.http.javadsl.server.Unmarshaller
+import java.util.Optional
+import akka.http.javadsl.server.Route
+import akka.http.scaladsl.common.ToNameReceptacleEnhancements.string2NR
+import akka.http.scaladsl.server.directives.ParameterDirectives._
+import scala.compat.java8.OptionConverters._
+import scala.collection.JavaConverters._
+import java.util.function.{ Function ⇒ JFunction }
+import java.util.{ Map ⇒ JMap, List ⇒ JList }
+import java.util.AbstractMap.SimpleImmutableEntry
+
+abstract class FormFieldDirectives extends FileAndResourceDirectives {
+
+  def formField(name: String, inner: JFunction[String, Route]): Route = ScalaRoute(
+    D.formField(name) { value ⇒
+      inner.apply(value).toScala
+    })
+
+  def formFieldOptional(name: String, inner: JFunction[Optional[String], Route]): Route = ScalaRoute(
+    D.formField(name.?) { value ⇒
+      inner.apply(value.asJava).toScala
+    })
+
+  def formFieldList(name: String, inner: JFunction[java.util.List[String], Route]): Route = ScalaRoute(
+    D.formField(string2NR(name).*) { values ⇒
+      inner.apply(values.toSeq.asJava).toScala
+    })
+
+  def formField[T](t: Unmarshaller[String, T], name: String, inner: JFunction[T, Route]): Route = {
+    import t.unwrap
+    ScalaRoute(
+      D.formField(name.as[T]) { value ⇒
+        inner.apply(value).toScala
+      })
+  }
+
+  def formFieldOptional[T](t: Unmarshaller[String, T], name: String, inner: JFunction[Optional[T], Route]): Route = {
+    import t.unwrap
+    ScalaRoute(
+      D.formField(name.as[T].?) { value ⇒
+        inner.apply(value.asJava).toScala
+      })
+  }
+
+  def formFieldList[T](t: Unmarshaller[String, T], name: String, inner: JFunction[java.util.List[T], Route]): Route = {
+    import t.unwrap
```

Comment:
I see why making it implicit was useful now... perhaps harmless and we can keep it so.